pkg/trie/cache/inmemory/trie_cache.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package inmemory
+
+import (
+	"github.com/ChainSafe/gossamer/pkg/trie/cache"
+)
+
+// https://github.com/paritytech/polkadot-sdk/blob/a8f4f4f00f8fc0da512a09e1450bf4cda954d70d/substrate/primitives/trie/src/cache/mod.rs#L98
+const defaultNodeCacheMaxSize = 8 * 1024 * 1024  // 8MB
+const defaultValueCacheMaxSize = 2 * 1024 * 1024 // 2MB
```

Comment:
I think it would be fine to use `ccache` to track number of items instead of bytes.  I don't think we need to adhere to the same strict size requirements for the node and value caches.  That would mean we wouldn't call `cacheValue.Size()` as often as well.

Reply:
If that's the case, we can reuse our current LRU cache implementation that @EclesioMeloJunior (https://github.com/ChainSafe/gossamer/pull/3895#discussion_r1594361263) a few days ago. 
My only concern is that we don’t know what that number could be and the size could vary. By keeping our current implementation, we can ensure we are not using more than `maxSize` bytes and remain compliant with the parity implementation.

Reply:
I think the `nodeCache` could definitely be counted by actual nodes instead of bytes and avoid doing the size checks.  For the `valueCache` it makes sense to track bytes and enforce a max limit.  I think this is a worthwhile change which differentiates our trie cache implementation for good reasons.